### ISSUE TITLE
Multiple remotes parallel search

### ISSUE THREAD

**Issue by iiknd** [conan-io/conan#5113]
Due to overall management of all our (upcoming) packages we need to split content into multiple repositories (Artifactory). Over the time the amount of repositories will be hundreds.

Is there any limit to conan client how many remotes it can handle?

`conan search -r all`

Does it search packages from each repository one by one? Is there a way to make it query servers in parallel, e.g. thread_count=16 or something similar?

**Comment by memsharded** [comment]
Hi @unzap 

Not, there is no limit. But yes, that command do a search in each repository, one by one, so it will take some time. So far there are no plans to parallelize it, as it hasn't been a big case yet. It is not that typical that users need to find a package in many multiple repositories. The possible first thing that will be parallelized is the upload, because that is more typical, and interesting time savings can be achieved.

So I think this is a bit early to be considered as a conan feature, I will classify it as a new feature, but low priority at the moment. If it is an important use case for you, there could be other approaches in the meantime:
- You might want to have a look to remotes and virtual repositories in Artifactory, as a way to gather and unify interface to different repositories. They have been released (without UI) in Artifactory 6.9, and with UI in 6.10.
- Implementing a small script that does a ``conan search -r=remote`` for each ``remote`` in parallel might be possible without a big effort

**Comment by AbrilRBS** [comment]
Closing as solved in Conan 2 with custom commands, which lets you overwrite Conan commands with your own custom implementations and manage them with `conan config install`

In this case, something like this would do the trick (Untested, just as an example):
_cmd_search.py_
```py
from collections import OrderedDict
from multiprocessing.pool import ThreadPool

from conan.api.conan_api import ConanAPI
from conan.api.model import ListPattern
from conan.cli.command import conan_command
from conan.cli.commands.list import print_list_text, print_list_json
from conans.errors import ConanException


@conan_command(group="Consumer", formatters={"text": print_list_text,
                                             "json": print_list_json})
def search(conan_api: ConanAPI, parser, *args):
    """
    Search for package recipes in all the remotes (by default), or a remote.
    """
    parser.add_argument('reference', help="Recipe reference to search for. "
                                          "It can contain * as wildcard at any reference field.")
    parser.add_argument("-r", "--remote", action="append",
                        help="Remote names. Accepts wildcards. If not specified it searches "
                             "in all the remotes")
    args = parser.parse_args(*args)
    ref_pattern = ListPattern(args.reference, rrev=None)
    if ref_pattern.package_id is not None or ref_pattern.rrev is not None:
        raise ConanException("Specifying a recipe revision or package ID is not allowed")

    remotes = conan_api.remotes.list(args.remote)
    if not remotes:
        raise ConanException("There are no remotes to search from")

    def _search_remote(remote):
        try:
            return {remote.name: conan_api.list.select(ref_pattern, package_query=None, remote=remote)}
        except Exception as e:
            return {remote.name: {"error": str(e)}}
    ##########
    # This is the custom part
    thread_pool = ThreadPool(len(remotes))
    results = thread_pool.map(_search_remote, remotes)
    thread_pool.close()
    thread_pool.join()

    return {
        "results": {remote: contents.serialize()
                    for result in results
                    for remote, contents in result.items()}
    }

```

If users keep requesting a similar feature we might think about adding it to the conan-extensions repository if we decide it makes sense :)
